### PR TITLE
Implementation For Withdrawal Related APIs

### PR DESCRIPTION
**Pull request by wharleyinc**
##Implementation

Added models for all withdrawal related data objects
Added methods in the RemoteCall interface to facilitate withdrawal requests
Added an example package to test remote calls

[https://docs.ovalfi.com/docs/initiate-savings-withdrawal](url)
##Paired With
CC @odetolakehinde

### CODE REVIEW

File: api/withdrawal.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/ovalfi/go-sdk/helpers"
+	"github.com/ovalfi/go-sdk/model"
+	"net/http"
```
Comment (odetolakehinde) [inline api/withdrawal.go:8]: not go formatted.

File: model/model.go [hunk 2]
```diff
@@ -1,7 +1,11 @@
 // Package model defines object and payload models
 package model
 
-import "database/sql"
+import (
+	"database/sql"
+	"github.com/google/uuid"
+	"time"
```
Comment (odetolakehinde) [inline model/model.go:7]: this too.

**Review by odetolakehinde — CHANGES_REQUESTED** [review]
Is this the only withdrawal function?

**Comment by wharleyinc** [comment]
> Is this the only withdrawal function?

According to the docs, we only have this. But if it is required, I can add other functions.

**Review by odetolakehinde — DISMISSED** [review]

**Review by azeezlala — APPROVED** [review]

File: api/withdrawal.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+const withdrawalAPIVersion = "v1/withdrawal"
+
+// InitiateWithdrawal makes an API request using Call to initiate a withdrawal
+func (c *Call) InitiateWithdrawal(ctx context.Context, request model.InitiateWithdrawalRequest) (model.Withdrawal, error) {
+	endpoint := fmt.Sprintf("%s%s", c.baseURL, withdrawalAPIVersion)
+
+	fL := c.logger.With().Str("func", "InitiateWithdrawal").Str("endpoint", endpoint).Logger()
+	fL.Info().Msg("starting...")
+	fL.Info().Str("customerID", request.CustomerID).Str("reference", request.Reference).
+		Float64("amount", request.Amount).
```
Comment (odetolakehinde) [inline api/withdrawal.go:21]: update using the interface method.

**Review by odetolakehinde — APPROVED** [review]

**Review by Dark-Vinci — APPROVED** [review]

**Pull request merged**